formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt

```diff
@@ -1864,5 +1866,64 @@ class FormulaRuntimeTest(val runtime: TestableRuntime, val name: String) {
 
         assertThat(batchScheduler.batchesOutsideOfScope).hasSize(0)
     }
+
+    @Test fun `batched events notify the inspector of start and stop`() {
+        val globalInspector = TestInspector()
+        FormulaPlugins.setPlugin(object : Plugin {
+            override fun inspector(type: KClass<*>): Inspector {
+                return globalInspector
+            }
+        })
+        val localInspector = TestInspector()
+
+        val childFormulaCount = 3
+
+        val batchScheduler = StateBatchScheduler()
+        val relay = runtime.newRelay()
+
+        val childFormula = IncrementActionFormula(
+            incrementRelay = relay,
+            executionType = Transition.Batched(batchScheduler)
+        )
+
+        val rootFormula = object : StatelessFormula<Unit, Int>() {
+            override fun Snapshot<Unit, Unit>.evaluate(): Evaluation<Int> {
+                val sum = (0 until childFormulaCount).sumOf { id ->
+                    context.key(id) {
+                        context.child(childFormula, input)
+                    }
+                }
+                return Evaluation(
+                    output = sum,
+                )
+            }
+        }
+
+        val subject = runtime.test(rootFormula, Unit, localInspector)
+        batchScheduler.performUpdate { relay.triggerEvent() }
+
+        // Inspect!
+        for (inspector in listOf(localInspector, globalInspector)) {
+            // Filtering out logs before "batch-started"
+            val events = inspector.events.dropWhile { !it.contains("batch-started") }
```

Comment:
nit: I wonder if we should pass `batchStartedEventPrefix` as `TestInspector` field - to decipher this hidden `"batch-started"` dependency?